examples/min-platform/embedding/wasmtime-platform.h

```diff
@@ -0,0 +1,203 @@
+ *
+ * Aborts the process on failure.
+ *
+ * Similar to `munmap` on Linux.
+ */
+extern void wasmtime_munmap(uint8_t *ptr, uintptr_t size);
+
+/**
+ * Configures the protections associated with a region of virtual memory
+ * starting at `ptr` and going to `size`.
+ *
+ * Aborts the process on failure.
+ *
+ * Similar to `mprotect` on Linux.
+ */
+extern void wasmtime_mprotect(uint8_t *ptr, uintptr_t size, uint32_t prot_flags);
+
+/**
+ * Returns the page size, in bytes, of the current system.
+ */
+extern uintptr_t wasmtime_page_size(void);
+
+/**
+ * Used to setup a frame on the stack to longjmp back to in the future.
+ *
+ * This function is used for handling traps in WebAssembly and is paried
+ * with `wasmtime_longjmp`.
+ *
+ * * `jmp_buf` - this argument is filled in with a pointer which if used
+ *   will be passed to `wasmtime_longjmp` later on by the runtime.
+ * * `callback` - this callback should be invoked after `jmp_buf` is
+ *   configured.
+ * * `payload` and `callee` - the two arguments to pass to `callback`.
+ *
+ * Returns 0 if `wasmtime_longjmp` was used to return to this function.
+ * Returns 1 if `wasmtime_longjmp` was not called an `callback` returned.
+ */
+extern int32_t wasmtime_setjmp(const uint8_t **jmp_buf,
+                               void (*callback)(uint8_t*, uint8_t*),
+                               uint8_t *payload,
+                               uint8_t *callee);
+
+/**
+ * Paired with `wasmtime_setjmp` this is used to jump back to the `setjmp`
+ * point.
+ *
+ * The argument here was originally passed to `wasmtime_setjmp` through its
+ * out-param.
+ *
+ * This function cannot return.
+ *
+ * This function may be invoked from the `wasmtime_trap_handler_t`
+ * configured by `wasmtime_init_traps`.
+ */
+extern void wasmtime_longjmp(const uint8_t *jmp_buf);
+
+/**
+ * Initializes trap-handling logic for this platform.
+ *
+ * Wasmtime's implementation of WebAssembly relies on the ability to catch
+ * signals/traps/etc. For example divide-by-zero may raise a machine
+ * exception. Out-of-bounds memory accesses may also raise a machine
+ * exception. This function is used to initialize trap handling.
+ *
+ * The `handler` provided is a function pointer to invoke whenever a trap
+ * is encountered. The `handler` is invoked whenever a trap is caught by
+ * the system.
+ */
+extern void wasmtime_init_traps(wasmtime_trap_handler_t handler);
+
+/**
+ * Attempts to create a new in-memory image of the `ptr`/`len` combo which
+ * can be mapped to virtual addresses in the future. The returned
+ * `wasmtime_memory_image` pointer can be `NULL` to indicate that an image
+ * cannot be created. The structure otherwise will later be deallocated
+ * with `wasmtime_memory_image_free` and `wasmtime_memory_image_map_at`
+ * will be used to map the image into new regions of the address space.
+ */
+extern struct wasmtime_memory_image *wasmtime_memory_image_new(const uint8_t *ptr, uintptr_t len);
```

Comment:
Is `ptr` guaranteed to stay valid until the memory image is freed or does the implementation need to copy the data elsewhere?